### PR TITLE
Update cmake

### PR DESCRIPTION
**Pull request by simonpintarelli**
- ~if exists libs/cxxopts is used as submodule, otherwise use find_package~
-  rely on cxxopts as external dependency
   - Readme: user must install cxxopts from brew/apt
   - Readme: `git clone` instead of `git cone --recursive`
- cmake use rocm provided by official `find_package(rocblas)`
- set TILEDDMM_GPU_BACKEND as option with a default value
- always install, remove TILEDMM_WITH_INSTALL cmake option
- use TARGET_NAME_IF_EXISTS in target_link_libraries
- Tiled-MMConfig add targets for tiledmm::cuda|rocm
- export compile_commands.json

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -42,23 +45,11 @@ if(TILEDMM_GPU_BACKEND STREQUAL "CUDA")
     list(APPEND TILEDMM_DEFINITIONS TILED_MM_CUDA)
 
 elseif(TILEDMM_GPU_BACKEND STREQUAL "ROCM")
-    find_package(HIPLIBS REQUIRED)
-    find_package(ROCBLAS REQUIRED)
-    list(APPEND TILEDMM_EXTERNAL_LIBRARIES ROCBLAS::rocblas HIPLIBS::hiplibs)
-    list(APPEND TILEDMM_DEFINITIONS -D__HIP_PLATFORM_HCC__) # since we are not using the hcc compiler, we have to set up this explicitly
+    find_package(rocblas REQUIRED)
+    list(APPEND TILEDMM_EXTERNAL_LIBRARIES rocs::rocblas)
```
Comment (kabicm) [inline CMakeLists.txt:49]: is `rocs` a typo for `rocm` or is it on purpose?

Comment (simonpintarelli) [reply CMakeLists.txt]: Thanks, yes this is a typo.

File: CMakeLists.txt [hunk 2]
```diff
@@ -38,72 +39,69 @@ if(TILEDMM_GPU_BACKEND STREQUAL "CUDA")
...
-                  "${Tiled-MM_SOURCE_DIR}/cmake/FindROCBLAS.cmake"
-                  "${Tiled-MM_SOURCE_DIR}/cmake/FindHIPLIBS.cmake"
-            DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Tiled-MM")
-endif()
+install(FILES "${Tiled-MM_BINARY_DIR}/Tiled-MMConfig.cmake"
+  "${Tiled-MM_BINARY_DIR}/Tiled-MMConfigVersion.cmake"
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Tiled-MM")
 
 if(TILEDMM_WITH_TESTS OR TILEDMM_WITH_EXAMPLES)
+  include(GitSubmodule)
```
Comment (mtaillefumier) [inline CMakeLists.txt:89]: we should probably remove cxxopts as submodule as we are not developing it. Treating this as external dependency is better.

**Review by mtaillefumier — APPROVED** [review]
I have no objection for merging this PR. the cxxopts comment is not blocking at all

File: CMakeLists.txt (hunk 2 above)
Comment (simonpintarelli) [reply CMakeLists.txt:89]: Oh, I forgot to remove it, it's also inconsistent with COSMA where cxxopts is a external-only dependency.

Comment (mtaillefumier) [reply CMakeLists.txt]: no worries

**Comment by mtaillefumier** [comment]
feel free to merge it whenever you think it is ready.

**Comment by simonpintarelli** [comment]
> feel free to merge it whenever you think it is ready.

Ok, let's wait until Marko approves it.

**Comment by kabicm** [comment]
Thanks Simon and Mathieu! Looks great to me!

**Review by kabicm — APPROVED** [review]

**Pull request merged**